dali/operators/generic/reshape.cc

```diff
@@ -247,10 +265,19 @@ void Reshape<Backend>::CalculateOutputShape(const Workspace &ws) {
   switch (shape_source_) {
     case ShapeSource::Arg:
       if (use_rel_shape_) {
+        if (!src_dims_.empty()) {
+          DALI_ENFORCE(rel_uniform_shape_.size() == src_dims_.size(),
+            make_string(OpName(), ": ``src_dims`` and ``rel_shape`` should have the same"
+            " length when both of them are provided. Got ", src_dims_.size(), " and ",
+            rel_uniform_shape_.size(), " elements respectively"));
```

Comment:
Not a must - just a suggestion. I think we can make it shorter ans skip the part about "when both are provided", becuase they obviously were, if we hit this error:
```suggestion
            make_string(OpName(), ": ``src_dims`` and ``rel_shape`` have different"
            " lengths: ", src_dims_.size(), " vs ", rel_uniform_shape_.size()));
```